client_plugin/drivers/shared/etcdops.go

```diff
@@ -280,3 +288,122 @@ func addrToEtcdClient(addr string) (*etcdClient.Client, error) {
 
 	return cli, nil
 }
+
+// Update or Create volume metadata in KV store
+func (d *VolumeDriver) writeVolMetadata(name string,
+	record VolumeMetadata) error {
+
+	var msg string
+	var value string
+	var err error
+
+	value = fmt.Sprintf("%d:", record.port)
+	value += record.username + ":"
+	value += record.password + ":"
+	for _, element := range record.clientList {
+		value += element
+		value += "?"
+	}
```

Comment:
yes, this is actually better. Please try to avoid sprintfs/etc and instead just keep json